test/e2e/e2e_test.go

```diff
@@ -54,6 +56,14 @@ func TestE2E(t *testing.T) {
 	})
 
 	cSuite.Setup(t)
+	addAdditionalResourcesToSuite(t, cSuite)
 	t.Logf("Running %d E2E tests", len(tests.ConformanceTests))
 	cSuite.Run(t, tests.ConformanceTests)
 }
+
+// set up additional resources that are created and cleaned up programmatically like certificates
+func addAdditionalResourcesToSuite(t *testing.T, testSuite *suite.ConformanceTestSuite) {
```

Comment:
It's not a strong opinion, could we just put the ConfigMap and Secrets in the yaml files for BackendTLSPolicy tests? They're not used elsewhere.

Reply:
at least, add a TODO

Reply:
I was trying to follow the patterns in upstream GW-API for these sorts of test (e.g. TLS Route). My only concern is that these certs will expire at some point and break tests, so I prefer to create them programatically...

Reply:
address this tought `My only concern is that these certs will expire at some point and break tests, so I prefer to create them programatically..`

Reply:
Created a static 100-year CA and leaf cert. I expect that these test will be redundant anyway once GW-API creates conformance for that. 
Do we have github secret scanner that needs to be update with an exclude?